hydra-cluster/src/HydraNode.hs

```diff
@@ -161,23 +162,50 @@ waitForAll tracer delay nodes expected = do
         _ ->
           tryNext c msgs stillExpected
 
+-- | Create a commit tx using the hydra-node for later submission
+externalCommit :: HydraClient -> UTxO -> IO Tx
+externalCommit HydraClient{hydraNodeId} utxos =
+  runReq defaultHttpConfig request
+    <&> responseBody
+    >>= \DraftCommitTxResponse{commitTx} -> pure commitTx
+ where
+  request =
+    Req.req
+      POST
+      (Req.http "127.0.0.1" /: "commit")
+      (ReqBodyJson (DraftCommitTxRequest{utxos} :: DraftCommitTxRequest Tx))
+      (Proxy :: Proxy (JsonResponse (DraftCommitTxResponse Tx)))
+      (Req.port $ 4000 + hydraNodeId)
+
 getMetrics :: HasCallStack => HydraClient -> IO ByteString
 getMetrics HydraClient{hydraNodeId} = do
-  response <-
-    failAfter 3 $ queryNode hydraNodeId
-  when (getResponseStatusCode response /= 200) $ failure ("Request for Hydra-node metrics failed :" <> show (getResponseBody response))
-  pure $ getResponseBody response
+  response <- failAfter 3 $ queryNode hydraNodeId
+  let respBody = Req.responseBody response
+  when (Req.responseStatusCode response /= 200) $
+    failure ("Request for Hydra-node metrics failed :" <> show respBody)
+  pure respBody
 
-queryNode :: Int -> IO (Response ByteString)
+queryNode :: Int -> IO Req.BsResponse
 queryNode nodeId =
-  parseRequest ("http://127.0.0.1:" <> show (6000 + nodeId) <> "/metrics") >>= loop
+  request >>= loop
  where
-  loop req =
-    httpBS req `catch` onConnectionFailure (loop req)
+  request =
+    pure $
+      runReq
+        defaultHttpConfig
+        ( Req.req
+            Req.GET
+            (Req.http "127.0.0.1" /: "metrics")
+            Req.NoReqBody
+            Req.bsResponse
+            (Req.port $ 6000 + nodeId)
+        )
+  loop r =
+    r `catch` onConnectionFailure (loop r)
 
   onConnectionFailure cont = \case
-    (HttpExceptionRequest _ (ConnectionFailure _)) -> threadDelay 100_000 >> cont
-    e -> throwIO e
+    (Req.VanillaHttpException _) -> threadDelay 100_000 >> cont
+    (Req.JsonHttpException _) -> threadDelay 100_000 >> cont
```

Comment:
This will loop forever in case the request is wrong. i.e. these are not only connection exceptions. I'd remove it and see if it works consistently in tests. If not, then we re-introduce connection retries here by matching on the `HttpException (HttpExceptionRequest _ (ConnectionTimeout))` or so